src/main/java/org/wildfly/security/auth/server/IdentityLocator.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server;
+
+import static org.wildfly.security._private.ElytronMessages.log;
+
+import java.security.Principal;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.evidence.Evidence;
+
+/**
+ * A locator for an identity.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class IdentityLocator {
```

Comment:
Is 'Locator' really a good name here?  To me locator gives the impression that it does something but from what I can see it is more a container for three optional values that can be used to obtain the realm identity.